lib/Dancer2/Core/Request.pm

```diff
@@ -181,6 +181,18 @@ sub data { $_[0]->{'data'} ||= $_[0]->deserialize() }
 sub deserialize {
     my $self = shift;
 
+    # if a serializer is used (JSON or Mutable) the deserialization of 
+    #   the whole 'multipart' section of the request is attempted
+    #   starting with the boundary
+    if (
+        $self->content_type 
+        && $self->content_type =~ /^multipart\/form-data/i 
+        && $self->serializer
```

Comment:
I think this last condition is not necessary. For one it's taken care of at line 196 right below, and for another I think the logical condition here is "no matter if we have a serializer or not, we don't no nuthing if the content type is `multipart/form-data`. Nuthing, I tell you.